samcli/commands/validate/validate.py

```diff
@@ -95,3 +114,40 @@ def _read_sam_file(template):
         sam_template = yaml_parse(sam_template.read())
 
     return sam_template
+
+def _lint(ctx, template):
+    """
+    Parses provided SAM template and maps errors from CloudFormation template back to SAM template.
+
+    Parameters
+    -----------
+    ctx
+        Click context object
+    template
+        Path to the template file
+
+    """
+
+    try:
+        if ctx.debug:
+            (args, filenames, formatter) = cfnlint.core.get_args_filenames([template, "--debug"])
+        else:
+            (args, filenames, formatter) = cfnlint.core.get_args_filenames([template])
+        LOGGER.setLevel(logging.WARNING)
+        matches = list(cfnlint.core.get_matches(filenames, args))
+        if not matches:
+            click.secho("{} is a valid SAM Template".format(template), fg="green")
+        rules = cfnlint.core.get_used_rules()
+        matches_output = formatter.print_matches(matches, rules, filenames)
+
+        if matches_output:
+            if args.output_file:
+                with open(args.output_file, 'w', encoding='utf-8') as output_file:
+                    output_file.write(matches_output)
+            else:
+                print(matches_output)
+
+        return cfnlint.core.get_exit_code(matches)
+    except cfnlint.core.CfnLintExitException as e:
+        LOGGER.error(str(e))
+        return e.exit_code
```

Comment:
Raise a `UserException` which will handle the error message and exit code, rather than manually returning the exit_code.

Reply:
will look into it, sounds good